### DISCUSSION TITLE
Project status

### DISCUSSION THREAD

**Discussion by freeboub** [TheWidlarzGroup/react-native-video#3350] [Announcements]
edited by @KrzysztofMoch 17/05/2024

# Project status

## Stable branch

Version **`6.0.0`** is the stable release (as of time of writing - May 2024).
`5.2.1` version is not maintained due to lack of contributors.

## Maintainers

Currently there are only 2 maintainers:
- @KrzysztofMoch (Krzysztof Moch) - [The Widlarz Group](https://thewidlarzgroup.com/). 
- @freeboub (Olivier Bouillet) - [iFeelSmart](https://ifeelsmart.com)

If anyone want to join the team you are welcome. You don't need to know all the package, being specialised in a subsystem is highly enough.

From time to time, we have some help some external contributors - Big thanks to you all 🙏

## Stable version - 6.0.0 (master branch)

Version 6.0.0 is introducing dozens of breaking changes, mostly through updated 
dependencies and significant refactoring. While the API remains compatible, the significant
 internal changes require full testing with your app to ensure all functionality remains operational.
Please view the [Changelog](https://github.com/react-native-video/react-native-video/blob/master/CHANGELOG.md) for specific breaking changes.

Regressions are tagged [critical for release](https://github.com/react-native-video/react-native-video/labels/critical%20for%20release), they must be fixed to consider this branch as stable.
If it suits you, you can use it in production

Main new feature:
- Ads insertion client side (google IMA)
- Typescript support
- Rewrite ios bidings in swift
- Upgrade exoplayer version (2.18)

Roadmap can be find here
- https://github.com/react-native-video/react-native-video/discussions/3351

## Communication

A discord channel react-native-video is available [here](https://discord.gg/WXuM4Tgb9X) 
But for sure, github is also a good way to trace changes and requests

**Comment by Firesoul18** [comment]
does this support drm protected videos with widevine as well?

**Reply by freeboub** [reply]
Yes

**Comment by jrhager84** [comment]
Is v6 considered 'production-ready?' If I can implement it into my project fairly easily, I'd be more than happy to contribute some time to fixing and doing PRs. I just have to develop against a project that has a tight deadline, and I would really want my efforts to align with this project. Are people having reliable success with the v6 beta release?

**Reply by gkueny** [reply]
Hi @jrhager84,

I do not think v6 is 'production-ready' as the "beta" tag indicates.

In my company we use it in **production** with success but it's not the main feature of our app. 
You should note that bug can happen & you need to test all your application's `react-native-video` functionality more carefully.

However, to weigh up the pros and cons, you need to know that v5 is no longer maintained.

**Reply by jrhager84** [reply]
That's my issue. I'd rather my work go to the RC, but I'm using 5.2 and I have to develop against my requirement, and it seems that there's a pretty significant departure from the old architecture and design.

**Reply by freeboub** [reply]
6.0.0 should go to RC soon. I think 5.2 is very buggy (even if it is tagged as stable). I really advise you to go with V6 and see what the blockers can be. Moreover backporting from 5.2 to 6.0 will be very difficult (especially for ios).
I understand it can be risky for a project, but more feedback we have better the V6 will be !

**Reply by freeboub** [reply]
@jrhager84 which kind of new requirement do you have ?

**Comment by Firesoul18** [comment]
can anyone give me a link for how to play the drm protected content on native player. Because I tried everything but it keeps giving me this error: ``{"error": {"extra": -2147483648, "what": 1}}``

I am sending the token and link as mentioned in this: https://docs.gumlet.com/reference/widewine-drm-with-react-native
as well as shown in different stack overflow questions and also tried the v6.0. It also gives the same error. Help me out guys

**Comment by afkcodes** [comment]
Hello @freeboub  do you have any update on this #3616 or is there any way to work around this ? Thank you
